server/record_test.go

```diff
@@ -0,0 +1,100 @@
+// +build record
+
+/*
+ * This file is part of Chihaya.
+ *
+ * Chihaya is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chihaya is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Chihaya.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package server
+
+import (
+	"bufio"
+	"math/rand"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+)
+
+type Record struct {
+	tid, uid  uint64
+	up, down  int64
+	absup     uint64
+	event, ip string
+}
+
+func randomString(len int) string {
+	bytes := make([]byte, len)
+	for i := 0; i < len; i++ {
+		bytes[i] = byte(65 + rand.Intn(90-65))
+	}
+	return string(bytes)
+}
+
+func randomIp() string {
```

Comment:
According to Go docs, this usage of Rand is deterministic and will produce same results on every run. This should be fixed by seeding it with time.Now() or something else, otherwise there's little point to it.